### PR TITLE
Enhancement/9621 update siwg client id url

### PR DESCRIPTION
**Pull request by jimmymadon**
## Summary

Addresses issue:

- #9621 

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 5.2 and PHP 7.4.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [ ] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.
- [ ] Ensure there are no unexpected significant changes to file sizes.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

File: assets/js/modules/sign-in-with-google/datastore/service.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+	 */
+	getClientIDURL: createRegistrySelector( ( select ) => () => {
+		const siteName = select( CORE_SITE ).getSiteName();
+		const homeURL = untrailingslashit( select( CORE_SITE ).getHomeURL() );
+		const supportEmail = select( CORE_USER ).getEmail();
+
+		const query = {
+			appname: siteName,
+			sitename: siteName,
+			siteurl: homeURL,
```
Comment (aaemnnosttv) [inline assets/js/modules/sign-in-with-google/datastore/service.js:86]: [Sorry for the change here](https://github.com/google/site-kit-wp/issues/9621#issuecomment-2477589188), but we need to pass the origin here instead. Essentially:
```suggestion
			siteorigin: new URL( homeURL ).origin,
```

File: assets/js/modules/sign-in-with-google/datastore/service.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+	),
+	/**
+	 * Gets a URL to fetch the client ID.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} state Data store's state.
+	 * @return {(string|undefined)} The URL to the clientID, or `undefined` if not loaded.
+	 */
+	getClientIDURL: createRegistrySelector( ( select ) => () => {
```
Comment (aaemnnosttv) [inline assets/js/modules/sign-in-with-google/datastore/service.js:78]: Not a bad idea to give this a dedicated selector but we generally name these using a `getService*URL` to indicate the URL is based on the serviceURL. Maybe `getServiceClientIDProvisioningURL` while a bit long would be much more clear :) WDYT?

File: assets/js/modules/sign-in-with-google/datastore/service.js [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+		const siteName = select( CORE_SITE ).getSiteName();
+		const homeURL = untrailingslashit( select( CORE_SITE ).getHomeURL() );
```
Comment (aaemnnosttv) [inline assets/js/modules/sign-in-with-google/datastore/service.js:80]: The trailing slash is irrelevant now given we only need the `origin` which wouldn't include any part of the path.

File: assets/js/modules/service-urls.test.js [hunk 4]
```diff
@@ -86,6 +86,11 @@ it( 'ensures all serviceURLs are properly constructed', () => {
 			'https://search.google.com/search-console/foo-path',
 			'https://search.google.com/search-console/foo-path?bar=baz',
 		],
+		'modules/sign-in-with-google': [
```
Comment (aaemnnosttv) [inline assets/js/modules/service-urls.test.js:89]: 👍

**Review by aaemnnosttv — CHANGES_REQUESTED** [review]
Thanks @jimmymadon – this looks great overall, just a few comments and [one change](https://github.com/google/site-kit-wp/issues/9621#issuecomment-2477589188) that just came in that makes more sense to do here than separately.

File: assets/js/modules/sign-in-with-google/datastore/service.js (hunk 2 above)
Comment (jimmymadon) [reply assets/js/modules/sign-in-with-google/datastore/service.js:78]: This sounds good to me!

**Review by aaemnnosttv — APPROVED** [review]
LGTM!

**Pull request merged**